plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBKeyValueStorageFactory.java

```diff
@@ -312,6 +320,18 @@ private Optional<DatabaseMetadata> handleVersionUpgrade(
     // In case we do an automated upgrade, then we also need to update the metadata on disk to
     // reflect the change to the runtime version, and return it.
 
+    // Besu supports both formats of receipts so no upgrade is needed other than updating metadata
+    if (runtimeVersion == BONSAI_WITH_RECEIPT_COMPACTION
+        || runtimeVersion == FOREST_WITH_RECEIPT_COMPACTION) {
+      final DatabaseMetadata metadata = new DatabaseMetadata(runtimeVersion);
+      try {
+        metadata.writeToDirectory(dataDir);
+        return Optional.of(metadata);
+      } catch (IOException e) {
+        throw new StorageException("Database upgrade to use receipt compaction failed", e);
+      }
+    }
+
```

Comment:
just reading the code the upgrade path looks good, just want to evaluate a special case, in case the user does this:

1. existing db metadata=v2
2. enables compact receipts -> metadata=v3
3. after sometime decides to disable compact receipts -> metadata remains v3, that make sense, since there are some compact receipts

Since the change apply only to new receipts, not sure the user expects/understands that the change applies to all receipts, do we want to prevent that, and trigger an error at point 3, telling it must resync?

Reply:
That's an interesting case. I think some kind of warning would be appropriate telling them that they have some existing compact receipts and if they want them all to be uncompacted they will need to resync but Besu will continue to function even if they don't. Don't like an error as seems like an artificial constraint and Besu will work fine with both receipt formats.

Reply:
Adding a warning seems reasonable to me

Reply:
done